internal/daemon/config.go

```diff
@@ -3,14 +3,20 @@ package daemon
 import (
 	"errors"
 	"github.com/creasty/defaults"
-	"github.com/goccy/go-yaml"
+	iglConfig "github.com/icinga/icinga-go-library/config"
```

Comment:
Personally, I'd try to avoid import aliases and rather rename the existing `config` variable instead. Could be something like `daemonConfig`, `globalConfig` or maybe just shorten it to `conf`.